src/base/bittorrent/torrentcreator.h

```diff
@@ -86,10 +91,10 @@ namespace BitTorrent
         void updateProgress(int progress);
 
     private:
-        void run() override;
         void sendProgressSignal(int currentPieceIdx, int totalPieces);
         void checkInterruptionRequested() const;
 
         TorrentCreatorParams m_params;
+        bool m_interruptionRequested = false;
```

Comment:
I don't think this is needed for the particular use-case. This serves as a shutdown signal. It can only be set to true. Even if many parallel threads write it at the same time, it will just end being set to true. 
However, if you think it's better for future-proofing, I can take care of switching to an atomic primitive.

Reply:
>However, if you think it's better for future-proofing, I can take care of switching to an atomic primitive.

Yes, I do. Also it signals that this variable is kind of special (modified between threads) not just a 'regular' variable.

Reply:
I used QAtomicInt for the task